### ISSUE TITLE
add ControlPanelNode/PlayAreaNode to ScreenView.js

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#381]
This name is a bloated way of asking, how do we want sims to use the a11y remapping implemented in https://github.com/phetsims/scenery-phet/issues/365 to look for phetsims? At the beginning of this issue, we were talking about putting "h2 level" nodes directly on the `ScreenView.js`. This could be pretty nice, especially with the remapping, and usages would look like:

```js
ScreenView.call( this);
this.playArea.accessibleOrder = [ . . . ];
this.controlPanel.accessibleOrder = [ . . . ];
```

Note, these nodes wouldn't have to be in the sim's scene graph at all. so the scene graph would look like:

```
rootNode
  simScene
     button1
     accordionBox
          button2
  playArea
```

Tagging @jonathanolson @jessegreenberg for thoughts.

**Comment by zepumph** [comment]
Marking as high priority since we should really know this sooner rather than later. My thought is to discuss this in an a11y meeting, then take it to all the devs for opinions.

**Comment by jonathanolson** [comment]
It feels like a fairly nice pattern to me.

> Note, these nodes wouldn't have to be in the sim's scene graph at all. so the scene graph would look like:

The playArea/controlPanel nodes definitely need to be in the scene graph (otherwise they wouldn't be tied to an accessible display).

**Comment by jonathanolson** [comment]
Also, let me know if there's a specific meeting where I should be available to discuss (I don't have strong preferences, so up to you).

**Comment by zepumph** [comment]
>The playArea/controlPanel nodes definitely need to be in the scene graph (otherwise they wouldn't be tied to an accessible display).

Agreed, sorry for confusion, I meant they could be tacked onto the root node directly, and didn't have to be included in any sim structure that was set up prior to a11y implementation. Basically they can be nodes that are siblings to the node tree structure for the sim view.

**Comment by zepumph** [comment]
This is beginning to be accomplished in https://github.com/phetsims/joist/issues/509 with `ScreenSummaryNode` (taking the place of `SceneSummaryNode`)

**Comment by zepumph** [comment]
Disussed in a11y dev today. We like that there is an option toggling the creation of the `screenSummaryNode`, and think that we should keep that option for the other two nodes as well. Eventually we want to have them all default to true, with the option to opt out, but for now, at least the screenSummaryNode needs to opt in, see https://github.com/phetsims/scenery-phet/issues/393 for changing usages of sceneSummary to this pattern.

**Comment by zepumph** [comment]
We want the names of the other nodes to be:
`playAreaNode`
`controlAreaNode`

**Comment by jessegreenberg** [comment]
After #365, this should generally not require changing the structure of the scene graph. Additional nodes on the `ScreenView` will allow us to specify a separate hierarchy without changing the scene graph, but in a consistent way since we expect this structure to show up frequently.
```js
ScreenView.call( this );

var otherNode = new Node();
var resetAllButton = new ResetAllButton();
this.addChild( resetAllButton )
this.addChild( otherNode );

this.playAreaNode.accessibleOrder = [ otherNode ];
this.controlAreaNode.accessibleOrder = [ resetAllButton ];
```

**Comment by jessegreenberg** [comment]
Adding developer meeting label since there will be strong opinions about this and this has long term impacts on a11y instrumentation. Current proposal for devs reading through this issue for the first time:

- ScreenView will have public Nodes:
  - `screenSummaryNode`
  - `playAreaNode`
  - `controlAreaNode`

Nodes can be added directly to these children if that is best. But generally, they will just be in place to control the structure of the scene graph, used like the example in https://github.com/phetsims/scenery-phet/issues/381#issuecomment-412610269. In that example, the ResetAllButton is added as a child of the ScreenView, but its accessible content will appear under the "Control Area" section.

**Comment by jonathanolson** [comment]
Sounds good to me. Do we run into any current name conflicts that will need to be handled? And can you overwrite their accessibleOrder, or would it already have things on it?

**Comment by zepumph** [comment]
> Do we run into any current name conflicts that will need to be handled?

I haven't run into any, and didn't see any in a project wide search just now.

>And can you overwrite their accessibleOrder, or would it already have things on it?

`playAreaNode` and `controlAreaNode` are just empty Nodes, but `screenSummaryNode` is different, and you cannot set it's `accessibleOrder`. 

We could state an overarching convention that you can't set the play/controlArea's accessibleOrder also, and just have to add node's to them instead, but that may be confusing and dumb.

**Comment by jonathanolson** [comment]
> We could state an overarching convention that you can't set the play/controlArea's accessibleOrder also, and just have to add node's to them instead, but that may be confusing and dumb.

That sounds too restrictive. Especially for older sims, I'd want to set the complete accessibleOrder of the node if possible.

**Comment by zepumph** [comment]
>I'd want to set the complete accessibleOrder of the node if possible.

I agree that is ideal. For `screenSummaryNode`, there is generic text for an intro that we want all sims to have (for consistency). Perhaps we can make it so that the public facing node is a child to a node with the intro in it, so that you could still set the `accessibleOrder` on it.

**Comment by samreid** [comment]
I'm confused about the example in https://github.com/phetsims/scenery-phet/issues/381#issuecomment-412610269

Does putting `otherNode` in `playAreaNode.accessibleOrder` mean it is no longer in the default order of `this`?

**Comment by jonathanolson** [comment]
> Does putting otherNode in playAreaNode.accessibleOrder mean it is no longer in the default order of this?

Yes. Generally, if you have a node parent/child and the child is in ANY accessibleOrder, it will not show up in the normal "tree" location.

**Comment by mbarlow12** [comment]
I generally like the approach proposed in https://github.com/phetsims/scenery-phet/issues/381#issuecomment-412639382 as well. My one question is how we'll handle some of the more complex screen summaries that we've seen thus far like RIAW? Would it be a new Node (possibly with children) than handles its own content updates that is then added as a child to the sim's `screenSummaryNode`? e.g.

**Comment by samreid** [comment]
I'm also wondering what happens if you don't add anything to `screenSummaryNode`, does assistive technology still visit it?

**Comment by zepumph** [comment]
https://github.com/phetsims/joist/issues/511 addresses the idea of validation for decisions made in this issue.

**Comment by jessegreenberg** [comment]
More notes from 8/16/18 dev meeting:

Overall devs had concern that over the design implications, that having two areas like "play area" and "control area" is not adequate for PhET simulations, and categorizing into these two sections is an arbitrary process. We do have the flexibility to add or change the names or structure if necessary in the future.

On the technical side, it was pointed out that *every* UI component in the ScreenView will need to be categorized into one of `controlAreaNode` or `playAreaNode`, because `<section>` is a block level element and acts as a container for content.

There was a request to add assertions or other verification that the screen hierarcy/order is correct. @zepumph made an issue to investigate.

We weren't able to come to a solid conclusion and want to come back to this later. We will put together an exmple of this being used and use that next time to guide the discussion further.

**Comment by jessegreenberg** [comment]
> I'm also wondering what happens if you don't add anything to screenSummaryNode, does assistive technology still visit it?

Yes, the general content in ScreenSummaryNode will be readable with an AT if you don't add more to it. Things like  "This screen is interactive, it changes as you play with it..." and so on.

> Would it be a new Node (possibly with children) than handles its own content

@mbarlow12 That's right.

**Comment by zepumph** [comment]
>I'm also wondering what happens if you don't add anything to screenSummaryNode, does assistive technology still visit it?

`screenSummaryNode` has default generic accessible content no matter what, so even if you didn't add anything else to it in the `{{Sim}}ScreenView.js` instance, you would still see something like "This is an interactive sim, it has x screens" or something.

**Comment by jessegreenberg** [comment]
Here is an example with the sections added to ScreenView, in a subtype called AccessibleScreenView:
https://github.com/phetsims/joist/blob/master/js/AccessibleScreenView.js

Here are two example usages in example-sim and masses-and-springs. Both cases are using `accessibleOrder` to  change the order and hierarchy of the PDOM without changing the scene graph. This is probably what will be done generally since it is convenient and less intrusive.
[Example Sim](https://github.com/phetsims/example-sim/blob/accessibleScreenView/js/example/view/ExampleScreenView.js)
[masses-and-springs: Control Area](https://github.com/phetsims/masses-and-springs/blob/accessibleScreenView/js/common/view/SpringScreenView.js#L249-L250)
[masses-and-springs: Play Area](https://github.com/phetsims/masses-and-springs/blob/accessibleScreenView/js/intro/view/IntroScreenView.js#L255-L270)

For next developer meeting hopefully  we can review these examples and any concerns, and decide whether we want to continue to use a subtype with this or whether these should be added directly to ScreenView.

**Comment by jessegreenberg** [comment]
>  each of this.screenSummaryNode, this.playAreaNode, this.controlAreaNode may be null. Is it OK to call setAccessibleOrder with an array with null elements? If so, is that a desirable API?

From `Accessibility.setAccessibleOrder:`
> The placeholder (`null`) will get filled in with all direct children that are NOT in any accessibleOrder. If there is no placeholder specified, it will act as if the placeholder is at the end of the order. The value `null` (the default) and the empty array (`[]`) both act as if the only order is the placeholder, i.e. `[null]`.

One result is that you can define that something should come last or first *and* last in the order with things like
```
aNode.accessibleOrder = [ null, lastNode ];
aNode.accessibleOrder = [ firstNode, null, lastNode ];
```

> the accessibleOrder is identical to the order added to the parent, so isn't that redundant?

Correct, this is the default order. But this line will ensure that these three are first in the PDOM if you do something like `screenView.insertChild();`. That isn't clear, sorry!

> I have recently been considering a dependency-injection approach...The given implementation is inflexible.

I agree. But I am not thrilled about requiring `ScreenSummaryNode`, `PlayAreaNode`, and `ControlAreaNode` in all sim ScreenViews. In this case, I don't think flexibility is necessary and the `options` should be removed from `ControlAreaNode` and `PlayAreaNode`.

> Why not add the barMagnetNode as a child of the playAreaNode, and the controlPanel as a child of the controlAreaNode?

That could also be done and would have the same result. IMO the flexibility is OK but we could consider defining a standard.

**Comment by jessegreenberg** [comment]
We discussed this in developer meeting on 8/30/18, and determined the following:
- [ ] PlayAreaNode, ControlAreaNode, and ScreenSummaryNode should be renamed to clearly indicate that they are not visual and are purely for a11y.
- [ ] Options should be removed from PlayAreaNode and ControlAreaNode, clients shouldn't be changing their tag names (for instance).

We discussed the following options:
1) Add `PlayAreaNode`, `ControlAreaNode`, and `ScreenSummaryNode` directly to ScreenView.
2) Add `PlayAreaNode`, `ControlAreaNode`, and `ScreenSummaryNode` to a subtype of ScreenView called `AccessibleScreenView`.
3) Just create and add new `PlayAreaNode`s, `ControlAreaNode`s and `ScreenSummaryNode`s to sim screen views as needed as part of the instrumentation process.

Devs didn't have strong preferences between these. We compared client usages between the options. Option 1 reduces a little bit of boilerplate in the a11y instrumentation while 3 is a little more flexible.  I don't think option 2 offers any benefit compared to options 1 and 3.

No matter what we choose, it would be very easy to change in the future, so developers gave the a11y team freedom to proceed as they see fit.

**Comment by zepumph** [comment]
On top of what was decided on in https://github.com/phetsims/scenery-phet/issues/381#issuecomment-417440555, we also need to make sure that sims aren't using custom processes, like in Faraday's law, and Molecules and Light, there are custom Nodes adding the heading.

**Comment by zepumph** [comment]
@jessegreenberg and I discussed this today, and thought that we should work on this prior to adding descriptions to Molarity. This would be nice to have a decided approach in place before molarity, just because it will be one less usage to convert later.

**Comment by zepumph** [comment]
Today @twant and I (but mainly @twant) had a good idea of trying not to add children to `ScreenView.screenSummaryNode` but instead passing a node up in the ScreenView constructor. 

This sounds much nicer and cleaner to me, and would change these lines in MolarityScreenView.js:

```js
    ScreenView.call( this, {
      layoutBounds: new Bounds2( 0, 0, 1100, 700 ),
      tandem: tandem,
      addScreenSummaryNode: true
    } );

    const molarityScreenSummaryNode = new MolarityScreenSummaryNode( model, valuesVisibleProperty );
    this.screenSummaryNode.addChild( molarityScreenSummaryNode );
```

into 
```js

    ScreenView.call( this, {
      layoutBounds: new Bounds2( 0, 0, 1100, 700 ),
      tandem: tandem,
      screenSummaryNode: new MolarityScreenSummaryNode( model, valuesVisibleProperty );
    } );

```

And if you need to add children later, you could create a container yourself, pass it up, and then addChild as you need to for that container:

```js

   const container = Node();

    ScreenView.call( this, {
      layoutBounds: new Bounds2( 0, 0, 1100, 700 ),
      tandem: tandem,
      screenSummaryNode: container
    } );

   container.addChild( /* other stuff */);
```

**Comment by zepumph** [comment]
@jessegreenberg and I discussed what we want to see `ScreenView.js` take on in terms of PDOM container nodes, as well as the idiomatic way to access them:

1. Add an option to screen view to pass in a screen summary node. Then ]`ScreenView.screenSummaryNode` no longer needs to be a member var.
2. Add `this.playAreaNode` and `this.controlAreaNode` to `ScreenView.js`. public and read only
3. Rules to structure the PDOM idiomatically:
    1. Don't convert children of the screen view over to be children of the PDOM containers (play area or control area)
    2. Instead use `accessibleOrder` to pluck those Nodes into the PDOM, without interfering with the view layout.
    3. Because of limitations outlined in https://github.com/phetsims/molarity/issues/63, every node needs to be in the scene graph, and not just specified by an `accessibleOrder`. We tried to make an assertion for this, but found it too strict in https://github.com/phetsims/scenery/issues/961.

Here is an example of how a screenView should look in terms of accessible content:

```js
class BlargScreenView extends ScreenView {

  constructor() {

    options = _.extend( {
      screenSummaryNode: new BlargScreenSummaryNode()
    }, options );

    // Nodes that were already there prior to adding a11y

    // create PDOM specific Nodes only needed for a11y 
    // add PDOM specific node to play area/control area

    // set accessible orders, this includes all nodes above that we want in the PDOM
    this.playAreaNode.accessibleOrder = [];
    this.controlAreaNode.accessibleOrder = [];

    // This likely won't change from prior to a11y instrumentation, the goal here is that we don't want useless
    // Nodes to interfere with what this is communicating: the order of the visual layout
    this.children = [
      /*
      Nodes that were already in the scene graph
       */
    ];


  }
}
```

**Comment by zepumph** [comment]
Ugh, there is one problem with that above:


Notice here in molarity that the screen summary node needs a Property created after options because it uses an option. In order to support this, I think we need to add a setter to the screenView.
```js
  function MolarityScreenView( model, tandem ) {

    this.model = model;

    ScreenView.call( this, {
      layoutBounds: new Bounds2( 0, 0, 1100, 700 ),
      tandem: tandem,
      addScreenSummaryNode: true
    } );

    const valuesVisibleProperty = new BooleanProperty( false, {
      tandem: tandem.createTandem( 'valuesVisibleProperty' )
    } );

    // a11y - creates screen summary in the PDOM and add it to the screenView
    const molarityScreenSummaryNode = new MolarityScreenSummaryNode( model, valuesVisibleProperty );
    this.screenSummaryNode.addChild( molarityScreenSummaryNode );

```

**Comment by zepumph** [comment]
I discussed a bit more with @jessegreenberg and updated the above list in https://github.com/phetsims/scenery-phet/issues/381#issuecomment-478731559

**Comment by zepumph** [comment]
Here is a patch I worked on today. It implemented just the new option for setting the screen summary node. I think it works alright, and it is worth pursuing more. For now I just implemented Friction, and didn't maintain backwards compatibility, so I will want to take this patch and convert all in one swoop so as not to move backwards (if we decide to go this direction). It is still a little weird to have this be an option and a setter, but maybe not too bad. So far I didn't implement an es5 setter for it.

<details>

```

Index: joist/js/Screen.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- joist/js/Screen.js	(revision 7613c005a1bb0ec7b70b9ecd5febb9915f328a3a)
+++ joist/js/Screen.js	(date 1554167144899)
@@ -296,12 +296,9 @@
           } );
         }
 
-        if ( this._view.screenSummaryNode ) {
-          assert && assert( this._view.screenSummaryNode instanceof ScreenSummaryNode, 'type different from expected, was it overwritten?' );
 
-          // if there is a screenSummaryNode, then set its intro string now
-          this._view.screenSummaryNode.setIntroString( simName, numberOfScreens );
-        }
+        // if there is a screenSummaryNode, then set its intro string now
+        this._view.setScreenSummaryIntroString( simName, numberOfScreens );
       }
       assert && this._view.accessibleAudit();
     },
Index: joist/js/ScreenView.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- joist/js/ScreenView.js	(revision 7613c005a1bb0ec7b70b9ecd5febb9915f328a3a)
+++ joist/js/ScreenView.js	(date 1554167255945)
@@ -33,7 +33,10 @@
 
       // a11y - TEMP option. to opt in rather than fix every sim structure right now, https://github.com/phetsims/joist/issues/509
       // This option is a placeholder right now as we convert old nodes to use this new screen summary structure,
-      addScreenSummaryNode: false
+      addScreenSummaryNode: false, // @deprecated
+
+      // {Node|null} - the Node with screen summary content to be added to the PDOM above the Play Area.
+      screenSummaryNode: null
     }, options );
     this.layoutBounds = options.layoutBounds;
 
@@ -54,18 +57,17 @@
     // @public (read-only)
     this.visibleBoundsProperty = new Property( options.layoutBounds );
 
-    // a11y - {Node|null}
-    //  This node is used to add an overview to the PDOM directly below the H1 (sim/screen name)
-    // only set to type {Node} if `includeScreenOverviewNode` is true.
-    // NOTE: DO NOT SET accessibleOrder ON THIS NODE
-    // @public (read-only) - you can add children to it
-    this.screenSummaryNode = null;
+    // @private
+    // a11y - this Node is suffixed "container" because it is added to with sim specific screen summary content, often
+    // called {Sim}ScreenSummaryNode. This container has the intro "{sim} is an interactive sim, it changes as you . . ."
+    this._screenSummaryNodeContainer = new ScreenSummaryNode();
+    this.addChild( this._screenSummaryNodeContainer );
 
-    if ( options.addScreenSummaryNode ) {
+    // @private {Node|null} - keep track of the summary node, so that if it is set more than once, the previous one can be removed
+    this._screenSummaryNode = null;
 
-      this.screenSummaryNode = new ScreenSummaryNode();
-      this.addChild( this.screenSummaryNode );
-    }
+    // at the Node from options in the same way that can be done at any time
+    options.screenSummaryNode && this.setScreenSummaryNode( options.screenSummaryNode );
   }
 
   joist.register( 'ScreenView', ScreenView );
@@ -123,6 +125,30 @@
       dispose: function() {
         Node.prototype.dispose.call( this );
         this.disposeScreenView();
+      },
+
+      /**
+       * Set the screen summary of this screen view
+       * @param {Node} node
+       */
+      setScreenSummaryNode: function( node ) {
+        assert && assert( node instanceof Node );
+        assert && assert( node !== this._screenSummaryNode, 'this is already the screen summary node content' );
+
+        this._screenSummaryNode && this._screenSummaryNodeContainer.removeChild( this._screenSummaryNode );
+
+        this._screenSummaryNode = node;
+        this._screenSummaryNodeContainer.addChild( node );
+      },
+
+      /**
+       * Set the screen summary Node intro string
+       * @param {string} simName
+       * @param {number} numberOfScreens
+       * @public (joist-internal)
+       */
+      setScreenSummaryIntroString: function( simName, numberOfScreens ) {
+        this._screenSummaryNodeContainer.setIntroString( simName, numberOfScreens );
       }
     },
 
Index: friction/js/friction/view/FrictionScreenView.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- friction/js/friction/view/FrictionScreenView.js	(revision da572d23ac348b22a89cf30e175a2f3781ad06d0)
+++ friction/js/friction/view/FrictionScreenView.js	(date 1554167415027)
@@ -58,15 +58,7 @@
    * @constructor
    */
   function FrictionScreenView( model, tandem ) {
-
     const self = this;
-    ScreenView.call( this, {
-      layoutBounds: new Bounds2( 0, 0, model.width, model.height ),
-      addScreenSummaryNode: true // opt into the generic screen overview strategy provided by ScreenView.js see https://github.com/phetsims/joist/issues/509
-    } );
-
-    // @private
-    this.model = model;
 
     // a11y initialize the describers for auditory descriptions and alerts.
     TemperatureIncreasingDescriber.initialize( model );
@@ -74,9 +66,15 @@
     BreakAwayDescriber.initialize( model );
     BookMovementDescriber.initialize( model );
 
-    // a11y
-    const frictionSummaryNode = new FrictionScreenSummaryNode( model, THERMOMETER_MIN_TEMP, THERMOMETER_MAX_TEMP );
-    this.screenSummaryNode.addChild( frictionSummaryNode );
+    const frictionScreenSummaryNode = new FrictionScreenSummaryNode( model, THERMOMETER_MIN_TEMP, THERMOMETER_MAX_TEMP );
+
+    ScreenView.call( this, {
+      layoutBounds: new Bounds2( 0, 0, model.width, model.height ),
+      screenSummaryNode: frictionScreenSummaryNode
+    } );
+
+    // @private
+    this.model = model;
 
     // add physics book
     this.addChild( new BookNode( model, physicsString, {
@@ -211,7 +209,7 @@
       TemperatureIncreasingDescriber.getDescriber().reset();
       BreakAwayDescriber.getDescriber().reset();
       BookMovementDescriber.getDescriber().reset();
-      frictionSummaryNode.updateSummaryString();
+      frictionScreenSummaryNode.updateSummaryString();
     };
   }
 
Index: scenery-phet/js/accessibility/nodes/ScreenSummaryNode.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- scenery-phet/js/accessibility/nodes/ScreenSummaryNode.js	(revision 6e2433f84dbf24334e569d2989e40d1aebf54c90)
+++ scenery-phet/js/accessibility/nodes/ScreenSummaryNode.js	(date 1554167090323)
@@ -57,6 +57,7 @@
      * filled in later on during initialization.
      * @param {string} simName
      * @param {number} numberOfScreens
+     * @public
      */
     setIntroString: function( simName, numberOfScreens ) {
 

```
</details>

**Comment by zepumph** [comment]
Above are commits that updated the API for setting the screen summary on a ScreenView. The new api supports passing in a Node via ScreenView options (`screenSummaryContent`). There is also a method to set it after the ScreenView call, `this.setScreenSummaryContent`, which worked really well in BASE, and wasn't needed anywhere else.

Next I will work on creating the playArea and controlArea nodes in the screen view.

**Comment by zepumph** [comment]
accessible section nodes added to ScreenView and usages updated in the above commits. I also deleted the unused `AccessibleSectionNode`. @jessegreenberg will you please review.

**Comment by zepumph** [comment]
accessible section nodes added to ScreenView and usages updated in the above commits. I also deleted the unused AccessibleSectionNode. @jessegreenberg will you please review.

**Comment by zepumph** [comment]
@jessegreenberg, while working on https://github.com/phetsims/gravity-force-lab/issues/141, I found that it was easy to blow away these PDOM Nodes defined in ScreenView.js when setting children for layout order (a pattern used in many sims).

As a result I would like to override setChildren in ScreenView. I did something like 

```
   setChildren( children ) {

        [ this._screenSummaryContent, this.playAreaNode, this.controlAreaNode ].forEach( pdomNode => {
          if ( children.indexOf( pdomNode ) < 0 ) {
            children.push( pdomNode );
          }
        } );
        Node.prototype.setChildren.call( this, children );
      },
```

But that caused a few ct problems, instead @pixelzoom mentioned that it would be better to house these Nodes in a parent, and then just add/remove them as a batch:

```
      setChildren( children ) {
        Node.prototype.setChildren.call( this, children );
        if ( !this.hasChild( this.pdomParent ) ) {
          this.addChild( this.pdomParent );
          this.pdomParent.moveToBack();
        }
      },
```
I'll give that a go.

**Comment by pixelzoom** [comment]
This broke multiple sims in CT today.  I asked in Slack:
> All of these PDOM nodes (_screenSummaryNode, playAreaNode, controlAreaNode) were added to ScreenView on 7/23/19.  Did anyone review this work? Did anyone review the work that was done yesterday?  Is a11y work that touches common code being code reviewed?

@zepumph replied:
> > Did anyone review this work?
> Still out for review in  https://github.com/phetsims/scenery-phet/issues/381

3+ months waiting for review of ScreenView changes seems excessive.

Raising priority of review to high.

**Comment by pixelzoom** [comment]
Comment that I made in Slack:

More on PDOM nodes in ScreenView…  Good chance that I would have unknowingly blown away `this.playAreaNode` in Natural Selection.  The entire design team refers to one of the UI elements (the area where the bunnies live) as the play area, and I was headed towards naming it playAreaNode.   I long ago gave up on the disconnect between how we generally use “play area” and how a11y uses it.  But it seems like it would be wise to put a prefix (“a11y”?) on all of the PDOM nodes that might be shadowed.

**Comment by zepumph** [comment]
My changes above are as follows:
* I moved PDOM Nodes to be a child of a single parent, and then added a `setChildren` override, as recommended by @pixelzoom, that conditionally adds the parent back in if it wasn't added during the `.children = [...]` call. Note that it shouldn't ever be part of that, as `pdomParent` is actually a private member, and there is no reason or need to access it from outside of ScreenView.js
* I added the `pdom` prefix to each PDOM Node, and changed usages
* I removed the prefixed underscore on screen summary related fields. 

@jessegreenberg ready for review.

**Comment by jessegreenberg** [comment]
This and changes after https://github.com/phetsims/scenery-phet/issues/381#issuecomment-547563528 look good @zepumph, thank you.

I also inspected usages in sims, I like this pattern. And I verified that order/structure in sims is correct with these Nodes. I think this can be closed.

**Comment by marlitas** [comment]
@pixelzoom recommended that I pass children into the ScreenView super constructor (https://github.com/phetsims/mean-share-and-balance/issues/44), however  @samreid and I ran into an assertion error. The assertion error was: `hasChild needs to be called with a Node`.

The lines in question were:
``` ts
public override setChildren( children: Node[] ): this {
    Node.prototype.setChildren.call( this, children );
    if ( !this.hasChild( this.pdomParentNode ) ) {
      this.addChild( this.pdomParentNode );
      this.pdomParentNode.moveToBack();
    }
    return this;
  }
```

**Comment by marlitas** [comment]
The above commit fixed this behavior. @jessegreenberg, can you review and if all seems good close?

**Comment by jessegreenberg** [comment]
This makes sense, thanks!
